### PR TITLE
Add associations

### PR DESCRIPTION
**Pull request by IsmaelMastronardi**
## This pull request adds the following changes to the codebase:

Created a Classroom class with a label attribute. Implemented getter and setter methods for label using attr_accessor. Established a has-many relationship between Classroom and Student, ensuring that adding a student to a classroom sets the classroom for the student and vice versa.

Introduced a Book class with title and author attributes and implemented getter and setter methods using attr_accessor.

Added a Rental class with a date attribute and implemented getter and setter methods.

Established a many-to-many relationship between Person and Book using the Rental class as an intermediate entity. Created the necessary associations for Book, Rental, and Person classes.

Here you have some code to test the app:

### Classroom and Person ***has-many/belongs-to*** relationship:

 #Create instances of Classroom
classroom1 = Classroom.new("Math Class")
classroom2 = Classroom.new("Science Class")

#Create instances of Student
student1 = Student.new(classroom1, 16, "Alice")
student2 = Student.new(classroom1, 17, "Bob")
student3 = Student.new(classroom2, 15, "Carol")

#Add students to classrooms
classroom1.add_student(student1)
classroom1.add_student(student2)
student3.add_classroom(classroom2)

#Test the relationship
puts "#{student1.name} is in #{student1.classroom.label}" # Output: Alice is in Math Class
puts "#{student2.name} is in #{student2.classroom.label}" # Output: Bob is in Math Class
puts "#{student3.name} is in #{student3.classroom.label}" # Output: Carol is in Science Class

### Person and Book  ***many-to-many *** relationship: 

#Create some books
book1 = Book.new("Book 1", "Author 1")
book2 = Book.new("Book 2", "Author 2")

#Create some persons
person1 = Person.new(25, "Alice")
person2 = Person.new(16, "Bob", parent_permission: true)
person3 = Person.new(30, "Charlie")

#Create rentals
rental1 = Rental.new("2023-10-01", book1, person1)
rental2 = Rental.new("2023-10-02", book2, person1)
rental3 = Rental.new("2023-10-03", book1, person2)
rental4 = Rental.new("2023-10-04", book2, person3)

#Check rentals for a person
puts "#{person1.name}'s Rentals:"
person1.rentals.each do |rental|
  puts "  Date: #{rental.date}, Book: #{rental.book.title}, Author: #{rental.book.author}"
end

puts "#{person2.name}'s Rentals:"
person2.rentals.each do |rental|
  puts "  Date: #{rental.date}, Book: #{rental.book.title}, Author: #{rental.book.author}"
end

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
```
Comment (rloterh) [inline book.rb]: - [ ] `rental` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ it. Kindly ensure to improve upon this aspect.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (rloterh) [inline book.rb]: - [ ] As required, let's ensure that a method for adding rental is present. The _book_ itself should be sent as a parameter to create the new _rental_, thereby achieving the _'has-many'_ association. We can achieve this by modifying our code to look like the following example below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :book, :person
```
Comment (rloterh) [inline rental.rb]: - [ ] Let's mark `book` and `person` with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ to it. Kindly ensure to improve upon this aspect.

File: person.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 require './nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (rloterh) [inline person.rb]: - [ ] Similarly, let's mark `rental` with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ it. Kindly ensure to improve upon this aspect.

File: person.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
   attr_accessor :age, :name
```
Comment (rloterh) [inline person.rb]: - [ ] Let's ensure a method for adding _rentals_ is present, so as to establish a a _rental_ `belong-to` _person_ relationship, like the code example below:

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ismael, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Good job so far with creating associations among objects!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by iambenkis — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- Good job so far with creating associations among objects! ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**